backends-velox/src/main/scala/org/apache/spark/sql/expression/UDFResolver.scala

```diff
@@ -152,34 +143,39 @@ object UDFResolver extends Logging {
 
   // Get the full paths of all libraries.
   // If it's a directory, get all files ends with ".so" recursively.
-  def getAllLibraries(files: String, sparkConf: SparkConf, canAccessSparkFiles: Boolean): String = {
+  def getAllLibraries(files: String, sparkConf: SparkConf): String = {
     val hadoopConf = SparkHadoopUtil.newConfiguration(sparkConf)
     files
       .split(",")
       .map {
         f =>
-          val file = new File(f)
-          // Relative paths should be uploaded via --files or --archives
-          // Use SparkFiles.get to download and unpack
-          if (!file.isAbsolute) {
-            if (!canAccessSparkFiles) {
-              throw new IllegalArgumentException(
-                "On yarn-client mode, driver only accepts absolute paths, but got " + f)
+          val uri = Utils.resolveURI(f)
```

Comment:
Looks like `f` can also be a relative filename or file tag here without a scheme. Would this logic be able to handle such case?

e.g.
```
--files /path/to/gluten/cpp/build/velox/udf/examples/libmyudf.so
--conf spark.gluten.sql.columnar.backend.velox.udfLibraryPaths=libmyudf.so
```

Reply:
yes，if f is `libmyudf.so`, Utils.resolveURI() will return `file://${PWD}/libmyudf.so`

Reply:
But `file://${PWD}/libmyudf.so` is not the expected path, right? It should be `/path/to/gluten/cpp/build/velox/udf/examples/libmyudf.so`

Reply:
The call to Utils.resolveURI() is used to determine whether the file is local or remote.

In your example, libmyudf.so is a local file, and we will not use uri.path directly.

The --files argument copies this file to a different destination directory. When --master=yarn is specified, the file is copied to the working directory on all nodes (both the driver and executors). In local mode, the files are added using SparkContext.addFile, and they can then be accessed using SparkFiles.get(f).

Reply:
> In your example, libmyudf.so is a local file

That doesn't make sense to me. `libmyudf.so` here should refer to the file uploaded by `--files`. But here it's resolved as a relative path to the runtime directory.

> The call to Utils.resolveURI() is used to determine whether the file is local or remote.

It would be better to check whether it's a relative path first. Although it's resolved as a local file and pass the `if` condition, the path doesn't even exist.

> When --master=yarn is specified, the file is copied to the working directory on all nodes (both the driver and executors).

Based on my previous experience, for yarn client mode, the files will be copied to all executor container + AM container, so they won't be copied to the driver node. In this case, if we only use the two configurations below
```
--files /path/to/gluten/cpp/build/velox/udf/examples/libmyudf.so
--conf spark.gluten.sql.columnar.backend.velox.udfLibraryPaths=libmyudf.so
```
the driver will fail to get the actual path for `libmyudf.so`, and that's the reason for adding  `spark.gluten.sql.columnar.backend.velox.driver.udfLibraryPaths`

Reply:
> It would be better to check whether it's a relative path first. Although it's resolved as a local file and pass the if condition, the path doesn't even exist.

Reply:
Thank you for your patient explanation, Rong! 

I previously thought that under yarn-client mode, `spark.yarn.dist.files` would also copy the files to the driver. I got the AM and driver mixed up.  

Let's fix this issue.

Reply:
I would suggest preserving 2 kinds of configuration for `spark.gluten.sql.columnar.backend.velox.udfLibraryPaths`

- relative path
- URI

And make `spark.gluten.sql.columnar.backend.velox.udfLibraryPaths=/path/to/..." invalid.

Reply:
@marin-ma  I have already fix the code according to the previous discussion in the comments, could you help review it again?